features/cloud.py

```diff
@@ -462,3 +493,156 @@ def _create_instance(
+            a image_name value
+        :param instance_name:
+            The name of the instance to be created
+        :param image_name:
+            The name of the image to be used when creating the instance
+        :param user_data:
+            The user data to be passed when creating the instance
+
+        :returns:
+            An AWS cloud provider instance
+        """
+        if not image_name:
+            image_name = self.locate_image_name(series)
+
+        image_type = self.name.title().replace("-", " ")
+
+        print(
+            "--- Launching {} image {}({})".format(
+                image_type, image_name, series
+            )
+        )
+
+        inst = self.api.launch(
+            name=instance_name, image_id=image_name, user_data=user_data
+        )
+        return inst
+
+    def get_instance_id(
+        self, instance: pycloudlib.instance.BaseInstance
+    ) -> str:
+        """Return the instance identifier.
+
+        :param instance:
+            An instance created on the cloud provider
+
+        :returns:
+            The string of the unique instance id
+        """
+        # For LXD, the API identifier uses the instance name
+        # instead of the instance id
+        return instance.name
+
+    def locate_image_name(self, series: str) -> str:
+        """Locate and return the image name to use for vm provision.
+
+        :param series:
+            The ubuntu release to be used when locating the image name
+
+        :returns:
+            A image name to use when provisioning a virtual machine
+            based on the series value
+        """
+        if not series:
+            raise ValueError(
+                "Must provide either series or image_name to launch azure"
+            )
+
+        image_name = self.api.daily_image(release=series)
+        return image_name
+
+
+class LXDVirtualMachine(_LXD):
+    name = "lxd-virtual-machine"
+
+    def manage_ssh_key(self, private_key_path: "Optional[str]" = None) -> None:
+        """Create and manage ssh key pairs to be used in the cloud provider.
+
+        :param private_key_path:
+            Location of the private key path to use. If None, the location
+            will be a default location.
+        """
+        pub_key_path = "lxd-pubkey"
+        priv_key_path = "lxd-privkey"
+        pub_key, priv_key = self.api.create_key_pair()
+
+        with open(pub_key_path, "w") as f:
+            f.write(pub_key)
+
+        with open(priv_key_path, "w") as f:
+            f.write(priv_key)
+
+        self.api.use_key(
+            public_key_path=pub_key_path, private_key_path=priv_key_path
+        )
+
+    @property
+    def api(self) -> pycloudlib.cloud.BaseCloud:
+        """Return the api used to interact with the cloud provider."""
+        if self._api is None:
+            self._api = pycloudlib.LXDVirtualMachine(tag=self.tag)
+
+        return self._api
```

Comment:
We might be able to define this simply in either in _LXD base class or even Cloud if we use a class attr to define the pycloublib.*Cloud callable to create

```diffdiff --git a/features/cloud.py b/features/cloud.py
index 38114a6..c2b8600 100644
--- a/features/cloud.py
+++ b/features/cloud.py
@@ -29,6 +29,7 @@ class Cloud:
     name = ""
     pro_ids_path = ""
     env_vars: "Tuple[str, ...]" = ()
+    pycloudlib_cls = None  # type: pycloudlib.cloud.BaseCloud
 
     def __init__(
         self,
@@ -64,7 +65,13 @@ class Cloud:
     @property
     def api(self) -> pycloudlib.cloud.BaseCloud:
         """Return the api used to interact with the cloud provider."""
-        raise NotImplementedError
+        if self.pycloudlib_cls is None:
+            raise NotImplementedError
+        if self._api is None:
+            self._api = self.pycloudlib_cls(
+                tag=self.tag, timestamp_suffix=self.timestamp_suffix
+            )
+        return self.api
 
     def manage_ssh_key(self, private_key_path: "Optional[str]" = None) -> None:
         """Create and manage ssh key pairs to be used in the cloud provider.
@@ -496,8 +503,6 @@ class Azure(Cloud):
 
 
 class _LXD(Cloud):
-    name = "_lxd"
-
     def _create_instance(
         self,
         series: str,
@@ -573,6 +578,7 @@ class _LXD(Cloud):
 
 class LXDVirtualMachine(_LXD):
     name = "lxd-virtual-machine"
+    pycloudlib_cls = pycloudlib.LXDVirtualMachine
 
     def manage_ssh_key(self, private_key_path: "Optional[str]" = None) -> None:
         """Create and manage ssh key pairs to be used in the cloud provider.
@@ -606,6 +612,7 @@ class LXDVirtualMachine(_LXD):
 
 class LXDContainer(_LXD):
     name = "lxd-virtual-machine"
+    pycloudlib_cls = pycloudlib.LXDContainer
 
     def manage_ssh_key(self, private_key_path: "Optional[str]" = None) -> None:
         """Create and manage ssh key pairs to be used in the cloud provider.
```